src/Npgsql/TypeHandlers/LqueryHandler.cs

```diff
@@ -0,0 +1,121 @@
+        {
+            if (lengthCache == null)
+                lengthCache = new NpgsqlLengthCache(1);
+            if (lengthCache.IsPopulated)
+                return lengthCache.Get() + 1;
+
+            // Add one byte for the prepended version number
+            return base.ValidateAndGetLength(value, ref lengthCache, parameter) + 1;
+        }
+
+        public override int ValidateAndGetLength(char[] value, ref NpgsqlLengthCache? lengthCache, NpgsqlParameter? parameter)
+        {
+            if (lengthCache == null)
+                lengthCache = new NpgsqlLengthCache(1);
+            if (lengthCache.IsPopulated)
+                return lengthCache.Get() + 1;
+
+            // Add one byte for the prepended version number
+            return base.ValidateAndGetLength(value, ref lengthCache, parameter) + 1;
+        }
+
+        public override int ValidateAndGetLength(ArraySegment<char> value, ref NpgsqlLengthCache? lengthCache, NpgsqlParameter? parameter)
+        {
+            if (lengthCache == null)
+                lengthCache = new NpgsqlLengthCache(1);
+            if (lengthCache.IsPopulated)
+                return lengthCache.Get() + 1;
+
+            // Add one byte for the prepended version number
+            return base.ValidateAndGetLength(value, ref lengthCache, parameter) + 1;
+        }
+
+        public override async Task Write(string value, NpgsqlWriteBuffer buf, NpgsqlLengthCache? lengthCache, NpgsqlParameter? parameter, bool async)
+        {
+            if (buf.WriteSpaceLeft < 1)
+                await buf.Flush(async);
+            buf.WriteByte(LqueryProtocolVersion);
+            await base.Write(value, buf, lengthCache, parameter, async);
+        }
+
+        public override async Task Write(char[] value, NpgsqlWriteBuffer buf, NpgsqlLengthCache? lengthCache, NpgsqlParameter? parameter, bool async)
+        {
+            if (buf.WriteSpaceLeft < 1)
+                await buf.Flush(async);
+            buf.WriteByte(LqueryProtocolVersion);
+            await base.Write(value, buf, lengthCache, parameter, async);
+        }
+
+        public override async Task Write(ArraySegment<char> value, NpgsqlWriteBuffer buf, NpgsqlLengthCache? lengthCache, NpgsqlParameter? parameter, bool async)
+        {
+            if (buf.WriteSpaceLeft < 1)
+                await buf.Flush(async);
+            buf.WriteByte(LqueryProtocolVersion);
+            await base.Write(value, buf, lengthCache, parameter, async);
+        }
+
+        #endregion
+
+        #region Read
+
+        public override async ValueTask<string> Read(NpgsqlReadBuffer buf, int len, bool async, FieldDescription? fieldDescription = null)
+        {
+            await buf.Ensure(1, async);
+            var version = buf.ReadByte();
+            if (version != LqueryProtocolVersion)
+                throw new NotSupportedException($"Don't know how to decode lquery with wire format {version}, your connection is now broken");
```

Comment:
Now it isn't. The driver properly handles invalid data in columns without breaking connections. By the way, we have `ThrowHelper` where you can put this code and avoid duplication in six places.

Reply:
As this handler is almost a one to one copy the wording should be fixed in these other places as well.
https://github.com/npgsql/npgsql/blob/main/src/Npgsql/TypeHandlers/JsonHandler.cs#L266

Reply:
Sure, will address this too (: